tiledb/api/src/array/fragment_info.rs

```diff
@@ -0,0 +1,667 @@
+                    },
+                )?;
+                unsafe {
+                    ffi::tiledb_fragment_info_get_non_empty_domain_var_size_from_index(
+                        self.context.capi(),
+                        self.capi(),
+                        frag_idx,
+                        c_dim_idx,
+                        &mut start_size,
+                        &mut end_size,
+                    )
+                }
+            }
+            LookupKey::Name(name) => {
+                let c_name = cstring!(name);
+                unsafe {
+                    ffi::tiledb_fragment_info_get_non_empty_domain_var_size_from_name(
+                        self.context.capi(),
+                        self.capi(),
+                        frag_idx,
+                        c_name.as_ptr(),
+                        &mut start_size,
+                        &mut end_size,
+                    )
+                }
+            }
+        })?;
+
+        let mut start = vec![0u8; start_size as usize].into_boxed_slice();
+        let mut end = vec![0u8; end_size as usize].into_boxed_slice();
+
+        self.capi_return(match dim_key.into() {
+            LookupKey::Index(idx) => {
+                let c_dim_idx: u32 = idx.try_into().map_err(
+                    |e: <usize as TryInto<u32>>::Error| {
+                        crate::error::Error::InvalidArgument(anyhow!(e))
+                    },
+                )?;
+                unsafe {
+                    ffi::tiledb_fragment_info_get_non_empty_domain_var_from_index(
+                        self.context.capi(),
+                        self.capi(),
+                        frag_idx,
+                        c_dim_idx,
+                        start.as_mut_ptr() as *mut std::ffi::c_void,
+                        end.as_mut_ptr() as *mut std::ffi::c_void,
+                    )
+                }
+            }
+            LookupKey::Name(name) => {
+                let c_name = cstring!(name);
+                unsafe {
+                    ffi::tiledb_fragment_info_get_non_empty_domain_var_from_name(
+                        self.context.capi(),
+                        self.capi(),
+                        frag_idx,
+                        c_name.as_ptr(),
+                        start.as_mut_ptr() as *mut std::ffi::c_void,
+                        end.as_mut_ptr() as *mut std::ffi::c_void,
+                    )
+                }
+            }
+        })?;
+
+        Ok(NonEmptyDomain::new(datatype, Range::from((start, end))))
+    }
+
+    pub fn mbr_num(&self, frag_idx: u32) -> TileDBResult<u64> {
+        let mut num: u64 = 0;
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_get_mbr_num(
+                self.context.capi(),
+                self.capi(),
+                frag_idx,
+                &mut num,
+            )
+        })?;
+
+        Ok(num)
+    }
+
+    pub fn mbr<T: Into<LookupKey> + Clone>(
```

Comment:
What does "mbr" mean?
Good time to start adding some method documentation :)

Reply:
Totes fair. For reference, mbr stands for [minimum bounding rectangle](https://en.wikipedia.org/wiki/Minimum_bounding_rectangle). Its from geospatial stuff. In our case it comes from the RTree used by sparse arrays for dimension indexing.

Reply:
With the convenience stuff I ended up creating a new type named `MinimumBoundingRectangle` for this.